pkg/provisioner/ironic/clients/auth.go

```diff
@@ -0,0 +1,95 @@
+package clients
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// AuthType is the method of authenticating requests to the server
+type AuthType string
+
+const (
+	// NoAuth uses no authentication
+	NoAuth AuthType = "noauth"
+	// HTTPBasicAuth uses HTTP Basic Authentication
+	HTTPBasicAuth AuthType = "http_basic"
+)
+
+// AuthConfig contains data needed to configure authentication in the client
+type AuthConfig struct {
+	Type     AuthType
+	Username string
+	Password string
+}
+
+func (auth *AuthConfig) load(clientType string) error {
+	authPath := path.Join("/auth", clientType)
```

Comment:
What is this doing? Where does "/auth" come from? Maybe this PR just also needs to update documentation?